browser/ui/webui/settings/brave_default_extensions_handler.cc

```diff
@@ -394,3 +404,16 @@ void BraveDefaultExtensionsHandler::SetBraveWalletEnabled(
   }
 }
 #endif
+
+void BraveDefaultExtensionsHandler::IsUnstoppableDomainsEnabled(
+    const base::ListValue* args) {
+  CHECK_EQ(args->GetSize(), 1U);
+  AllowJavascript();
+  ResolveJavascriptCallback(
+      args->GetList()[0],
+#if BUILDFLAG(UNSTOPPABLE_DOMAINS_ENABLED)
+      base::Value(unstoppable_domains::IsUnstoppableDomainsEnabled()));
```

Comment:
I'm thinking that if move this function (and the feature) to the browser layer, we would be able to get rid of several spots with `BUILDFLAG(UNSTOPPABLE_DOMAINS_ENABLED)`. Typically we want to have as few guards as possible

Reply:
Personally I would like to keep this util function in component layer unless we indeed need browser stuff, I think it makes it easier to use and include in the places we needed it.
And I think we would usually need to include things from unstoppable domains component in the places we use this check, which would still need the build flag.
This here might be the only case that we don't need other stuff in UD component.

Reply:
ok with me, maybe @bridiver would like to have a look at build guards